### PR TITLE
SignIn:FE

### PR DESCRIPTION
**Pull request by GabisPapin**
created redux, added react-hook-form, added styled-component. Obs: change signInPage to be return in layout file because in app return error.

### CODE REVIEW

File: src/components/signIn/Signin.styles.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import styled from 'styled-components';
+
+const backgroundColor = {
+  button: 'rgb(50, 90, 199)',
+  border: 'rgb(153, 152, 152)',
+}
+
+const backgroundForm = 'rgb(189, 197, 204)';
+
+const color = {
```
Comment (lyudad) [inline src/components/signIn/Signin.styles.ts:10]: it should be general constant  file for full app

File: src/components/signIn/Signin.styles.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  paragraphColor: 'rgb(61, 59, 59)',
+  buttonText: 'white',
+}
+
+const font = {
```
Comment (lyudad) [inline src/components/signIn/Signin.styles.ts:15]: it should be general constant  file for full app

File: src/redux/reducers/signin.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+interface signInState {
+  email: string,
+}
+
+const initialState: signInState = {
+  email: '',
```
Comment (lyudad) [inline src/redux/reducers/signin.ts:8]: why we need it?

File: src/redux/reducers/signin.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+export const signInSlice = createSlice({
+  name: 'signIn',
+  initialState,
+  reducers: {
+    email: (state, actions) => {
```
Comment (lyudad) [inline src/redux/reducers/signin.ts:15]: why we need it?

**Comment by GabisPapin** [comment]
Corrected the style file, created connection with signIn BE, created a forgot password component.

**Comment by GabisPapin** [comment]
Forgot password screen:
forgot-passowrd

**Comment by GabisPapin** [comment]
created constants/styles and put all color and font size. Obs: Didn't change the screen design.

File: src/service/signinHttp.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const authApi: any = createApi({
+  reducerPath: 'auth',
+  baseQuery: fetchBaseQuery({ baseUrl: 'http://localhost:3000' }),
```
Comment (lyudad) [inline src/service/signinHttp.ts:6]: no url in the code. or env or constants

Comment (GabisPapin) [reply src/service/signinHttp.ts]: I created a .env file.

**Comment by GabisPapin** [comment]
Created .env file and resolved all conflicts.

**Pull request merged**